common/src/main/scala/aws/AWSAsync.scala

```diff
@@ -6,9 +6,9 @@ import java.util.concurrent.{Future => JFuture}
 import scala.concurrent.{Future, Promise}
 
 object AWSAsync {
-  private def promiseToAsyncHandler[Request <: AmazonWebServiceRequest, Result](p: Promise[Result]) =
+  private def promiseToAsyncHandler[Request <: AmazonWebServiceRequest, Result](p: Promise[Result], request: AmazonWebServiceRequest) =
     new AsyncHandler[Request, Result] {
-      override def onError(exception: Exception): Unit = { p.failure(exception); () }
+      override def onError(exception: Exception): Unit = {p.failure(new RuntimeException(s"Dynamo request failed $request", exception)); ()}
```

Comment:
This part is meant to be used by any AWS service, so I'm not too keen on tracing the Dynamo specific request here.

Could that be simply done at the time the AWSAsync is used? The future is failed with an exception, so it should be possible to have myFuture.recover { case NonFatal(e) => blah blah} and find a way to pass the request at that point?

Reply:
Changing this for notification only for now.